Приветствую, Анатолий!
По 3у заданию вопрос с пониманием самой сути задания:
Есть вероятность, что задача может быть решена не в том ключе, так как ТЗ мне показалось противоречивым.
В самом начале значится: "Создать собственный класс-исключение, который должен проверять содержимое
списка на наличие только чисел."
А ниже вторым подпунктом идёт: "Запрашивать у пользователя данные и заполнять список необходимо только числами."
Поэтому проверяю не содержимое списка, а данные, которые вводит пользователь.
И уже от сюда или ошибка, или аппендим список.
Ошибка не останавливает работу скрипта, который бесконечен до ввода STOP (в любой вариации).

Задачи 4,5,6 мне показалось логично объеденить в один скрипт исполнения. Попытался показать, что освоил из ООП.

Спасибо за основы Python - было круто!
